templates/infrastructure/environment/aws/vpc/terragrunt.hcl

```diff
@@ -54,4 +49,14 @@ inputs = {
   create_database_internet_gateway_route = true
 
   map_public_ip_on_launch = true
+
+  cidr_block = {{if .Spec.import_resource}}"{{ .Spec.cidr_block}}"{{else}}"10.0.0.0/16"{{end}}
+  public_subnet_cidr_blocks = {{if .Spec.import_resource}}[{{ range .Spec.public_subnet_cidr_blocks}}"{{.}}", {{end}}]{{else}}["10.0.1.0/24", "10.0.2.0/24", "10.0.3.0/24"]{{end}}
+  private_subnet_cidr_blocks = {{if .Spec.import_resource}}[{{ range .Spec.private_subnet_cidr_blocks}}"{{.}}", {{end}}]{{else}}["10.0.4.0/24", "10.0.5.0/24", "10.0.6.0/24"]{{end}}
+  database_subnet_cidr_blocks = {{if .Spec.import_resource}}[{{ range .Spec.database_subnet_cidr_blocks}}"{{.}}", {{end}}]{{else}}["10.0.7.0/24", "10.0.8.0/24", "10.0.9.0/24"]{{end}}
+  elasticache_subnet_cidr_blocks = {{if .Spec.import_resource}}[{{ range .Spec.elasticache_subnet_cidr_blocks}}"{{.}}", {{end}}]{{else}}["10.0.10.0/24", "10.0.11.0/24", "10.0.12.0/24"]{{end}}
```

Comment:
have them as variables and these can be default values for them. simpler.

also rename that variable to remove the `import_` prefix. It can be provided manually for advanced use cases.
The subnet count should be := len(public_subnet_cidr_blocks)

Reply:
We are not using subnet count in the module...

Reply:
this is dead code that can be removed as well then.
https://github.com/argonautdev/tf-modules/blob/main/modules/aws/vpc/variables.tf#L66